databricks/koalas/indexes.py

```diff
@@ -317,6 +317,62 @@ def is_object(self):
         """
         return is_object_dtype(self.dtype)
 
+    def dropna(self):
+        """
+        Return Index or MultiIndex without NA/NaN values
+
+        Examples
+        --------
+
+        >>> df = ks.DataFrame([[1, 2], [4, 5], [7, 8]],
+        ...                   index=['cobra', 'viper', None],
+        ...                   columns=['max_speed', 'shield'])
+        >>> df
+               max_speed  shield
+        cobra          1       2
+        viper          4       5
+        NaN            7       8
+
+        >>> df.index.dropna()
+        Index(['cobra', 'viper'], dtype='object')
+
+        Also support for MultiIndex
+
+        >>> midx = pd.MultiIndex([['lama', 'cow', 'falcon'],
+        ...                       [None, 'weight', 'length']],
+        ...                      [[0, 1, 1, 1, 1, 1, 2, 2, 2],
+        ...                       [0, 1, 1, 0, 1, 2, 1, 1, 2]])
+        >>> s = ks.Series([45, 200, 1.2, 30, 250, 1.5, 320, 1, None],
+        ...               index=midx)
+        >>> s
+        lama    NaN        45.0
+        cow     weight    200.0
+                weight      1.2
+                NaN        30.0
+                weight    250.0
+                length      1.5
+        falcon  weight    320.0
+                weight      1.0
+                length      NaN
+        Name: 0, dtype: float64
+
+        >>> s.index.dropna()  # doctest: +SKIP
+        MultiIndex([(   'cow', 'weight'),
+                    (   'cow', 'weight'),
+                    (   'cow', 'weight'),
+                    (   'cow', 'length'),
+                    ('falcon', 'weight'),
+                    ('falcon', 'weight'),
+                    ('falcon', 'length')],
+                   )
+        """
+        kdf = self._kdf.copy()
```

Comment:
@itholic can we reuse DataFrame.dropna impl?

Reply:
Not pretty sure, i think they are slightly different when they filter the null value

In case DataFrame.dropna, they assume `column_index` as index,

```python
idxes = self._internal.column_index
reduce(lambda x, y: x + y,
       [F.when(self[idx].notna()._scol, 1).otherwise(0)
       for idx in idxes]
```

but in case Index.dropna, we assume `index_scols` as index.
```python
index_scol.isNotNull() for index_scol in kdf._internal.index_scols
```

Reply:
I see. can we use Spark's `DataFrame.dropna` API?

Reply:
ping @itholic can we use Spark's DataFrame.dropna API?

Reply:
@HyukjinKwon yeah i think we can simply use that one. thanks !! 👍